src/server/DeviceServer.js

```diff
@@ -327,6 +319,37 @@ class DeviceServer {
     await this._eventPublisher.publish(eventData);
   };
 
+  _onDeviceClaimCodeMessage = async (
+    message: Message,
+    device: Device,
+  ): Promise<void> => {
+    const claimCode = message.getPayload().toString();
+    const deviceID = device.getID();
+    const deviceAttributes = await this._deviceAttributeRepository.getById(deviceID);
+    if (!deviceAttributes) {
+      return;
+    }
+    if (deviceAttributes.ownerID) {
+      return;
+    }
+
+    // todo if we figure out how to delete claimCode from the device
+    // we could make the flow more clean.
+    if (deviceAttributes.claimCode !== claimCode) {
+      const claimRequestUser = await this._userRepository.getByClaimCode(claimCode);
+      if (!claimRequestUser) {
+        return;
+      }
+
+      await this._userRepository.removeClaimCode(claimRequestUser.id, claimCode);
```

Comment:
yes maybe it's not a bad idea. you need to access them only the first time the device reconnect after a setup, there are no other moments you need them. 
i don't know why i was sure that they should have been saved in user object...